### ISSUE TITLE
Error: This application can not open files.

### ISSUE THREAD

**Issue by bhaumik-tripathi** [drkrssll/chunks-rs#1]
Hello, I have been trying to create a parser for my "chuncks" But after all the work, I was getting the mentioned error( In the title), while trying to reproduce the error, this is the smallest working example which gives the error 
```
(r_widgets:121860): GLib-GIO-CRITICAL **: 20:26:33.519: This application can not open files.
```

CODE:
Move the contents in `main` function in the first example in readme file of this project into a function `open` which takes no input/ output, then change`main` function into following
```
fn main() {
    let args: Vec<String> = env::args().collect();
    if args[1] == "storage" {
        println!("{:?}", args[1]);
        open();
    }
}

//Contents of open

fn open() {
    let factory = Factory::new("chunk.factory");
    let chunks = |factory: GtkApp| {
        storage(&factory);
        load_css(STYLE);
    };
    factory.pollute(chunks);
}

```

Run the command:
```
cargo run -- storage
```
And you will get the error
```
(r_widgets:121860): GLib-GIO-CRITICAL **: 20:26:33.519: This application can not open files.
```
I think I maybe at fault somewhere, but have been debugging it for few hours without success.

**Comment by drkrssll** [comment]
I've been able to replicate this on my end. It doesn't seem as if you are at fault, but rather it seems to be an issue with the GTK4 Application not being configured to handle CLI arguments.

I'm working on a resolution, thanks for letting me know.

**Comment by bhaumik-tripathi** [comment]
This is not the write place, but since there are no Discussions in the repo or any discord, I am compelled to ask it here . 
`utils` have a `tag_box` which returns a `Tag` enum, but I ain't seeing any box in `Internal` (like say other internals like `static_button`) but seems `Bar` 's description matches the definition of box(as in eww), it has orientation and can take other `utils`/widgets inside it, so is `Bar` to be used as a box/container widget and also can I parse a `Bar` inside another `Bar`?
Previously I was thinking that it is something made specifically for making bars.

**Comment by drkrssll** [comment]
To try and cure your confusion about the Tags - the Tag enum was just a simple way for me to handle GTK's widget types. Think of the Tag enum as a list of potential GTK4 widgets. By returning a Tag instead of a Box, I can easily handle different widget types via a match statement:

```rs
let child = match tag {
    Tag::Label(label) => label.clone().upcast::<Widget>(),
    Tag::Box(box_) => box_.clone().upcast::<Widget>(),
    Tag::Button(button) => button.clone().upcast::<Widget>(),
};
```

It may seem a bit weird at first, but this helped tremendously when adding functionality to GTK4 widgets without having to make new functions for each.

As for your question about Bars, I'm fairly certain that you can't parse a Bar into another Bar, as Bars are not only a gtk4 Box, but primarily an ApplicationWindow. From what I can see on reddit at least, ApplicationWindows can't be used inside ApplicationWindows without some external library.

If you would like to toy around with Chunks to help better suit your use case, by all means you are welcome to submit a pull request. I would very much appreciate the help.

**Comment by drkrssll** [comment]
@bhaumik-tripathi I just released version 0.7.0 which allows for argument parsing! Here is a code example to help you get started:

```rs
fn main() {
    let factory = Factory::new("chunk.factory");
    let args: Vec<String> = env::args().collect();

    let command_line_handler = |app: &GtkApp, cmd_line: &GtkCmdLine| {
        let args = cmd_line.arguments();
        if args.len() < 2 {
            eprintln!("No command provided");
            return 1;
        }

        match args[1].as_os_str() {
            os if os == OsStr::new("storage") => {
                storage(app);

                load_css(STYLE);
                0
            }
            os if os == OsStr::new("clock") => {
                clock(app);

                load_css(STYLE);
                0
            }
            _ => {
                eprintln!("Unknown argument: {}", args[1].to_string_lossy());
                1
            }
        }
    };

    factory.pollute_with_args(command_line_handler, args);
}
```

It seems a bit much at first glance, but this is the only way I could manage to get it to work. Closing this issue now, feel free to reach out to me on [X/twitter](https://x.com/drkrssll) if you'd like to keep up our conversation.

**Comment by bhaumik-tripathi** [comment]
thanks @drkrssll , I will take a look and let you know! It would be quite helpful if you could open Discussions on the repo, so that discussions related to repo stay together with project. I don't use X.
